testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/xalan/XalanUsageTestCase.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ *
+ */
+
+package org.jboss.as.test.integration.xalan;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.URL;
+
+/**
+ * A testcase to ensure that user deployments can package and use xalan library of their own
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class XalanUsageTestCase {
+
+    private static final String WEB_APP_CONTEXT = "xalan-webapp";
+
+
+    @ArquillianResource
+    private URL appURL;
+
+    @Deployment
+    public static EnterpriseArchive createDeployment() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, WEB_APP_CONTEXT + ".war");
+        war.addClasses(XalanUsageServlet.class);
+        // add a dummy xml to transform
+        war.addAsResource(XalanUsageServlet.class.getPackage(), "dummy.xml", XalanUsageServlet.XML_RESOURCE_TO_TRANSFORM);
+        war.addAsResource(XalanUsageServlet.class.getPackage(), "dummy.xsl", XalanUsageServlet.XSL_RESOURCE);
+
+        final EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "xalan-usage.ear");
+        // add the .war
+        ear.addAsModule(war);
+        // add the xerces jar in the .ear/lib
+        ear.addAsLibrary(XalanUsageTestCase.class.getResource("xalan-2.7.2.jar"), "xalan.jar");
```

Comment:
Is there any way we could resolve these files from maven, rather than packaging the binaries?